### ISSUE TITLE
Default version of p7zip-full is outdated

### ISSUE THREAD

**Issue by QKaiser** [IoT-Inspector/unblob#131]
At the moment we rely on the default version of 7zip that is available as `p7zip-full` in our Linux distributions.

These versions are quite outdated, case in point with latest 7z version on Ubuntu 20.04 and our Docker container (python3.8-slim):

```
7-Zip [64] 16.02 : Copyright (c) 1999-2016 Igor Pavlov : 2016-05-21
```

While the latest version is:

```
7-Zip (z) 21.06 (x64) : Copyright (c) 1999-2021 Igor Pavlov : 2021-11-24
```

This is problematic for a few reasons:
- we lack potential enhancements from 7z's team on decompression/extraction speed
- we might miss security patches from 7z latest, although they may get backported by distribution maintainers (but we should not rely on that)
- we lack support for specific file formats (**I can't send a pull request for ext[2-4] because my version of 7z does not support ext4 extraction**)

### Solution: download and install 7z manually

7zip only provides a download page and there is no link to get the latest version automatically so we would need to keep track of it or parse the history file.

To get the (current) latest version of 7zip as a stripped binary:

```
curl -s https://www.7-zip.org/a/7z2106-linux-x64.tar.xz --output - | tar Jxvf - 7zzs
```

Of course we can use dumb tricks like this:

```
LATEST_VERSION=$(curl -s https://www.7-zip.org/history.txt | head -n 4 | tail -n 1 | awk '{ print $1 }' | sed 's/\.//')
curl -s "https://www.7-zip.org/a/7z${LATEST_VERSION}-linux-x64.tar.xz --output - | tar Jxvf - 7zzs
```

But I'm 100% sure it will break at some point if we do this.

**Comment by QKaiser** [comment]
To be discussed but I'd be in favor of either distributing 7zip in [unblob/extractors](https://github.com/IoT-Inspector/unblob/tree/main/unblob/extractors), or to have an install script somewhere in the repository that must be run prior to use unblob (could it be done with `poetry install` ?)

I think we're good on the licensing side (see https://www.7-zip.org/license.txt).

For our docker container it's just a line to add to the Dockerfile to fetch and install the latest version.

**Comment by QKaiser** [comment]
Can be closed as soon as we land #158
